tests/fixtures/images/validation_os_images.py

```
@@ -36,35 +40,49 @@ def validation_os_images_namespace(admin_client):
-        role_ref_name="view",
+        role_ref_name=OS_IMAGES_EDIT_CLUSTER_ROLE,
     )
 
+    # A RoleBinding created by an older revision of this fixture may still point to a different
+    # ClusterRole (e.g. "view"). A RoleBinding's roleRef is immutable, so a stale binding must be
+    # deleted and recreated with the correct roleRef; otherwise recreating it below fails with a
+    # 409 AlreadyExists conflict.
+    if role_binding.exists and role_binding.instance.roleRef.name != OS_IMAGES_EDIT_CLUSTER_ROLE:
+        LOGGER.info(
+            f"Deleting stale RoleBinding {role_binding.name} in {role_binding.namespace} "
+            f"pointing to {role_binding.instance.roleRef.name}, expected {OS_IMAGES_EDIT_CLUSTER_ROLE}"
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)